Updates Block docs to reflect changes in 1.15

Iuhegi opened the pull request.

Commit:
Updates Block docs to reflect changes in 1.13

williewillus commented on docs/blocks/blocks.md:
```diff
@@ -8,17 +8,30 @@ Creating a Block
+
+!!! Note
+
+    Blocks have no setter for Item Group (formerly Creative Tab). This has been moved to the ItemBlock, and is now it's responsibility. Furthermore, there is no setter for translation key (this is generated based on registry name now).
```
should be `its` (no ')

williewillus commented on docs/blocks/blocks.md:
```diff
@@ -8,17 +8,30 @@ Creating a Block
+    Blocks have no setter for Item Group (formerly Creative Tab). This has been moved to the ItemBlock, and is now it's responsibility. Furthermore, there is no setter for translation key (this is generated based on registry name now).
+
+### `Block.Builder`
+To set the properties of the block, you need to use a `Block.Builder`. There are a couple of options when creating a one. You can use `Block.Builder::from` to copy the properties of an existing block or `Block.Builder::create` to create a new one.
```
use `.` instead of `::`, people might get confused since `::` is used only for method refs in Java

Iuhegi replied:
good point

williewillus commented on docs/blocks/blocks.md:
```diff
@@ -8,17 +8,30 @@ Creating a Block
+  - `lightValue` - The amount of light emitted by the block. Example Usage: Glowstone. **Note:** this method takes a value from 1 to 15.
+  - `hardnessAndResistance` - Hardness is how long it takes to mine a block (set to -1 for unbreakable). Resistance is explosion resistance.
+  - `needsRandomTick` - makes the block recieve random ticks. Example Usage: Plants.
+  - `variableOpacity` - if set to true, the game will cache whether your block is a full cube in each state, and do stuff accordingly. Example Usage: Shulker Boxes
```
needs to be clearer, and i think your logic is backwards: if `variableOpacity` is set to true, the game *won't* cache the results of getLightOpacity (or whatever it's called) and will re-query every time.

Iuhegi replied:
oh, i must of gotten switched around when trying to understand the logic

Commit:
Ammends to fix problems

tterrag1098 commented:
This needs updates for 1.13.2 class name changes.

tterrag1098 commented on docs/blocks/blocks.md:
```diff
@@ -8,17 +8,30 @@ Creating a Block
+  - `doesNotBlockMovement` - makes it so the block does not block movement, Example Usage: plants.
+  - `slipperiness` - defaults to `0.6F` high values make it more slippery. Example Usage: Ice. **Note:** Vanilla minecraft does not exceed `1.0F` in their blocks slipperiness.
+  - `sound` - Defaults to `SoundType.STONE` see [sounds][] for more information.
+  - `lightValue` - The amount of light emitted by the block. Example Usage: Glowstone. **Note:** this method takes a value from 1 to 15.
```
0 to 15.

tterrag1098 commented on docs/blocks/blocks.md:
```diff
@@ -8,17 +8,30 @@ Creating a Block
+
+The `Block.Builder` has the following setters:
+
+  - `doesNotBlockMovement` - makes it so the block does not block movement, Example Usage: plants.
```
```suggestion
  - `doesNotBlockMovement` - makes it so the block does not block movement. Example Usage: plants
```

tterrag1098 commented on docs/blocks/blocks.md:
```diff
@@ -8,17 +8,30 @@ Creating a Block
+  - `lightValue` - The amount of light emitted by the block. Example Usage: Glowstone. **Note:** this method takes a value from 1 to 15.
+  - `hardnessAndResistance` - Hardness is how long it takes to mine a block (set to -1 for unbreakable). Resistance is explosion resistance.
+  - `needsRandomTick` - makes the block recieve random ticks. Example Usage: Plants.
+  - `variableOpacity` - if set to true, the game will work out the opacity of your block everytime either `IBlockstate.getOpacity`, `IBlockState.propagatesSkylightDown` or `IBlockState.getLightOpacity` is called. Example Usage: Shulker Boxes.
```
Capitalization of these bullet points is all over the place, be consistent.

LukeGi commented:
Busy atm with uni, if anyone else wants to finish this, i don't plan to for a couple weeks

kashike commented on docs/blocks/blocks.md:
```diff
@@ -8,17 +8,30 @@ Creating a Block
+
+    Blocks have no setter for Item Group (formerly Creative Tab). This has been moved to the ItemBlock, and is now its responsibility. Furthermore, there is no setter for translation key (this is generated based on registry name now).
+
+### `Block.Builder`
```
`Builder` -> `Properties`, in entire document.

brandonbothell committed:
Fix minor grammar issue in gettingstarted (#228)

tterrag1098 committed:
Merge branch 'master' into 1.13.x

tterrag1098 committed:
Update docs versions

tterrag1098 committed:
Set version to 1.15.x

tterrag1098 committed:
Fix version branch names

tterrag1098 committed:
Merge branch '1.14.x' into 1.15.x

Cadiboo committed:
Delete old information about `TileEntityRendererFast` (#251)

Aeltumn committed:
Updated Getting Started section (#239)

* Fix files link and add note to update run configurations

* Updated 'Structuring Your Mod' to 1.13+

* Forgot one field in mods.toml

* Can't update much about forge update checker as I don't use it myself

Did remove 2 dead links.

* Added examples back to update checker doc

* Applied suggestions by Cadiboo

* Added note that forge 1.14.4 is v28

Aeltumn committed:
Updated Blocks section (#244)

* Fully updated blocks section to 1.13+

* Made requested changes

Aeltumn committed:
Update Rendering section (#243)

* Updated rendering category

* Update abbreviation

kaltinril committed:
Change assets to data.  Add note about common vanilla format for filename (#255)

* Change assets to data.  Add note about common vanilla format for filename.

* Remove double-space after sentence ending.

kaltinril committed:
Add link to Forge provided tags list (#257)

Aeltumn committed:
Updated Events section (#242)

* Add more details about Mod.EventBusSubscriber properties

* Added more information about the mod event bus

* Minor rephrasing

* Added more information about the mod event bus

JoJoDeveloping committed:
Add documentation for writing side-only mods or network channels. (#240)

Because people are often not aware of their existance

Cadiboo committed:
Update Item to use Item.Properties (#252)

* Update for Item.Properties

* Update items.md

Cadiboo committed:
Registries: Small fixes + add info about DeferredRegister (#249)

* EntityEntry -> EntityType & add TileEntities + DeferredRegister

* Update registries.md

realmayus committed:
Fixed typo in example mods.toml (#263)

tterrag1098 committed:
Downgrade pymdown-extensions to avoid dependency issues

marvin-roesch committed:
Update MkDocs for Python 3 compatibility and fix search (#264)

tterrag1098 committed:
Remove outdated information

tterrag1098 committed:
Merge branch '1.14.x' into 1.15.x

tterrag1098 committed:
remove forge blockstates doc

Cadiboo committed:
Fix TileEntity references for 1.14+ (#250)

* Fix references for 1.13+

* Update tileentity.md

* Update tileentity.md

Liam-Ryan committed:
Update edit marker in index.md (#265)

tonitch committed:
typo correction (#273)

Cadiboo committed:
Getting Started: Add info about generating Run/Launch Configs (#247)

Cadiboo committed:
Update autoupdate.md (#248)

Aeltumn committed:
Added a legacy category, addresses #254 and #216 (#256)

* Added a legacy category, #254, #216

* Added some newlines

tterrag1098 committed:
Merge remote-tracking branch 'upstream/1.14.x' into 1.15.x

TheGreyGhost committed:
Updated Environment Setup instructions for IntelliJ (#260)

* Updated Environment Setup instructions for IntelliJ

The gradle tasks have been changed since the original instructions were written.  IDEA 2019 also performs most of the setup without manual tweaks required.

* Updated in response to pull request review

Draco18s committed:
Docs for global loot modifiers (#290)

mrp-v2 committed:
fix typo (worm -> warm) (#314)

TBBYT committed:
Update index.md (#301)

Fixed missing spaces before "*". (The bullet points do not display at https://mcforge.readthedocs.io/en/latest/gettingstarted/)

tterrag1098 committed:
Fix documentation on logoFile

Supersedes #300

sigmasoldi3r committed:
Removed deprecated link (#295)

As of #294 this link should be removed from the index.

Cyborgmas committed:
Add a datapack page (#292)

* Added a datapack page

* Modified the page and added references in other pages

* Applied suggested changes

Co-authored-by: tterrag <tterrag1098@gmail.com>

* More changes

* Finished refactoring and corrected conflict

* tidied some references

* Fix some grammar, remove unnecessary content

Co-authored-by: tterrag <tterrag1098@gmail.com>

ChampionAsh5357 committed:
1.15.x Changes (#323)

* Changes specific to 1.15.x

* Update color parameters

IBlockAccess -> ILightReader

* 1.14.x isn't legacy yet

Co-authored-by: tterrag <tterrag1098@gmail.com>

sciwhiz12 committed:
Add documentation for Access Transformers (#321)

* Add documentation for Access Transformers

* Integrate suggestions, fix a few paragraphs

* Add paragraph on refreshing gradle

sciwhiz12 committed:
Rewrite the docs for Block States (#288)

* Rewrite the docs for Block States

* Remove unneeded word

* Change to javadoc convention

* Integrate suggestions from review

Cyborgmas committed:
Particle Overview (#291)

* Particle explanation

Added a new page on how to use particles

Co-authored-by: tterrag <tterrag1098@gmail.com>

sciwhiz12 committed:
Data generators documentation (#311)

* Add introduction to data generators

* Add paragraphs on runData task, fix spacing

* Move paragraph on GatherDataEvent futher up

* List out the runs generation tasks

* Start work on model provider docs

* Integrate reviews, simplify some phrases, specify overridable methods

* Remove words referencing the reader

tterrag1098 committed:
Clean up README, add CONTRIBUTING file

tterrag1098 committed:
Merge remote-tracking branch 'upstream/1.14.x' into 1.15.x

# Conflicts:
#	docs/models/color.md

sciwhiz12 committed:
[1.14.x] Rewrite docs for Registries (#297)

* Rewrite registries documentation

* Move ObjectHolder section to middle, fix typos, add ContainerType

* Update phrasing

* Generalize note on registering *Type classes

* Fix typo

* Remove paragraph about registry load order

* Update references to registries docs

* Use new instances, not fields in registering example

* Convert and fix tabs to spacing

* Clarify ObjectHolder rules

* Clarify when ObjectHolder fields are refreshed

* Change phrase suggesting ResourceLocations are auto-lowercased

* Add section on RegistryObjects

* Expand phrase on Register event

Cyborgmas committed:
Fix references in particle page (#326)

tterrag1098 committed:
Switch active versions to 1.15/1.16

tterrag1098 committed:
Merge branch '1.14.x' into 1.15.x

# Conflicts:
#	docs/concepts/registries.md

TheCurle committed:
Remove reference to the FG Cookbook (#345)

sciwhiz12 commented:
Could you please rebase your PR to 1.15.x and adjust as suggested by the reviewers?
_(If you do not wish to continue this PR, please respond soon.)_

sciwhiz12 committed:
Move and update the docs on lifecycle events (#327)

* Move GatherDataEvent to proper position above Common Setup

* Add link to sides documentation

MichaelHillcox committed:
Updated TextComponentTranslation to TranslationTextComponent (#331)

ravenbuilder934 committed:
Broken link fixes (#352)

* Autoupdate link fix

* Fix tags link

* Remove link to "own blockstate format"

* Remove "item blockstates" link, remove unneeded parentheses

* TileEntityItemStackRenderer should be ItemStackTileEntityRenderer

* Remove extended blockstates link (page does not exist)

* Fix datapacks link

* Fix datapacks link on Tags page

Iuhegi commented:
I can work on this on friday :)

Iuhegi committed:
Rebase and Fix problems outlines by reviewers

Iuhegi commented:
i think i did this wrong
